botocore/parsers.py

```diff
@@ -964,11 +964,14 @@ def _parse_non_payload_attrs(
                     member_shape, headers
                 )
             elif location == 'header':
-                header_name = member_shape.serialization.get('name', name)
-                if header_name in headers:
-                    final_parsed[name] = self._parse_shape(
-                        member_shape, headers[header_name]
-                    )
+                self._parse_headers(final_parsed, headers, member_shape, name)
+
+    def _parse_headers(self, final_parsed, headers, member_shape, name):
```

Comment:
nit: I would use the singular version of this name (i.e. `_parse_header`) since this function only parses a single header.